src/commands/connection.ts

```diff
@@ -0,0 +1,68 @@
+import { SongQueue } from '@/utils/song-queue';
+import {
+  createAudioPlayer,
+  getVoiceConnection,
+  joinVoiceChannel,
+  VoiceConnectionStatus,
+} from '@discordjs/voice';
+import { CommandInteraction } from 'discord.js';
+import Container from 'typedi';
+
+const queue = Container.get(SongQueue);
+
+export function executeLeave(interaction: CommandInteraction) {
+  try {
+    getVoiceConnection(interaction.guildId).disconnect();
+    interaction.reply({
+      content: `Disconnected`,
+      ephemeral: true,
+    });
+  } catch (err) {
+    interaction.reply({
+      content: `I'm not connected`,
+      ephemeral: true,
+    });
+  }
+}
+
+export function executeJoin(interaction: CommandInteraction) {
+  const guild = interaction.guild;
+  const member = guild.members.cache.get(interaction.member.user.id);
+  const voiceChannel = member.voice.channel;
+
+  if (!voiceChannel) {
+    return interaction.channel.send(
+      'You must be in a voice channel to play music!'
+    );
+  }
+
+  if (!queue.getQueue(guild.id)) {
+    const queueObject = {
+      textChannel: interaction.channel,
+      voiceChannel,
+      connection: null,
+      player: createAudioPlayer(),
+      songs: [],
+      playing: true,
+    };
+    queue.setQueue(guild.id, queueObject);
+  }
+
+  queue.getQueue(guild.id).connection = joinVoiceChannel({
+    channelId: voiceChannel.id,
+    guildId: guild.id,
+    adapterCreator: guild.voiceAdapterCreator,
+  });
+
+  queue
+    .getQueue(guild.id)
+    .connection.on(VoiceConnectionStatus.Disconnected, () => {
+      console.log('Bot disconnected');
+      queue.getQueue(guild.id).connection.destroy();
+    });
+
+  interaction.reply({
```

Comment:
If we await the connection above, we should also use deferReply instead of reply

Reply:
abovementioned, there is no need for await

Reply:
![image](https://user-images.githubusercontent.com/13706500/134801129-f2d1adc1-46b1-4bd5-a1ba-d3adef42e9a7.png)
![image](https://user-images.githubusercontent.com/13706500/134801139-6c7db016-bc1d-4c4b-bafa-d090c1bebb61.png)
reply is promise btw

Reply:
No need to wait the reply, the problem is the bot has only 3 secs to respond to an interaction, otherwise it will show "Interaction fail". So if the connection command just fire and gone, it's fine, but if it blocks until it's connected successfully, we must use deferReply instead of normal reply. Accept for now but need to re-check this behavior.